pwndbg/gdblib/vmmap.py

```diff
@@ -204,7 +249,18 @@ def explore(address_maybe: int) -> pwndbg.lib.memory.Page | None:
         return None
 
     flags |= 2 if pwndbg.gdblib.memory.poke(address_maybe) else 0
-    flags |= 1 if not pwndbg.gdblib.stack.is_executable() else 0
+    # It's really hard to check for executability, so we just make some guesses:
+    # If it's in the same page as the stack pointer, try to check the NX bit
+    # If it's in the same page as the instruction pointer, assume it's executable
+    # Otherwise, just say it's not executable
+    if address_maybe == pwndbg.lib.memory.page_align(pwndbg.gdblib.regs.pc):
+        flags |= 1
+    # TODO: could maybe make this check look at the stacks in pwndbg.gdblib.stack.get() but that might have issues
+    elif (
+        address_maybe == pwndbg.lib.memory.page_align(pwndbg.gdblib.regs.sp)
+        and pwndbg.gdblib.stack.is_executable()
+    ):
+        flags |= 1 if pwndbg.gdblib.stack.is_executable() else 0
```

Comment:
This could be just `flags |= 1` since we already checked for `stack.is_executable()` in the `elif` condition.

Tho maybe this should be just `if addr == aligned(pc) or (addr == aligned(sp) and stack.is_executable()): flags |= 1`?